src/os/bluestore/BlueStore.cc

```diff
@@ -6021,8 +6029,8 @@ int BlueStore::fsck(bool deep)
 	  size_t next = used_blocks.find_next(cur);
 	  if (next != cur + 1) {
 	    derr << __func__ << " error: leaked extent 0x" << std::hex
-		 << ((uint64_t)start * block_size) << "~"
```

Comment:
We introduce these changes at https://github.com/ceph/ceph/pull/17268 to get the wrongly truncated last 4k block back.  Seems this won't be able to function as it is supposed to(after this patch is applied)...

Reply:
Yeah.. it occurs to me now that we actually have to revisit that fix, because it only corrects the problem if you run fsck, and that might not happen before they upgrade past luminous.  I think we should make mount() check detect and correct the condition.

Reply:
> I think we should make mount() check detect and correct the condition.

👍